platform/linux-generic/pktio/socket_xdp.c

```diff
@@ -0,0 +1,589 @@
+/* Copyright (c) 2022, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include <odp/autoheader_internal.h>
+
+#ifdef _ODP_PKTIO_XDP
+
+#include <odp/api/debug.h>
+#include <odp/api/hints.h>
+#include <odp/api/system_info.h>
+#include <odp/api/ticketlock.h>
+
+#include <odp_debug_internal.h>
+#include <odp_macros_internal.h>
+#include <odp_packet_io_internal.h>
+#include <odp_packet_internal.h>
+#include <odp_parse_internal.h>
+#include <odp_classification_internal.h>
+#include <odp_socket_common.h>
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <xdp/xsk.h>
+
+#define NUM_XDP_DESCS 1024U
+#define MIN_FRAME_SIZE 2048U
+
+typedef struct {
+	struct xsk_ring_prod fill_q;
+	struct xsk_ring_cons compl_q;
+	struct xsk_umem *umem;
+	pool_t *pool;
+} xdp_umem_info_t;
+
+typedef struct {
+	struct xsk_ring_cons rx;
+	struct xsk_ring_cons compl_q;
+	struct xsk_ring_prod tx;
+	struct xsk_ring_prod fill_q;
+	xdp_umem_info_t *umem_info;
+	struct xsk_socket *xsk;
+	int helper_sock;
+	uint32_t mtu;
+	uint32_t max_mtu;
+} xdp_sock_info_t;
+
+typedef struct {
+	odp_ticketlock_t rx_lock ODP_ALIGNED_CACHE;
+	odp_ticketlock_t tx_lock ODP_ALIGNED_CACHE;
+	xdp_umem_info_t umem_info;
```

Comment:
This member doesn't seem to be used anywhere.

Reply:
Leftover, removed in v9.